diff -up nbtk-1.2.3/docs/reference/libnbtk/Makefile.am.types nbtk-1.2.3/docs/reference/libnbtk/Makefile.am
--- nbtk-1.2.3/docs/reference/libnbtk/Makefile.am.types	2009-11-09 12:50:14.000000000 +0100
+++ nbtk-1.2.3/docs/reference/libnbtk/Makefile.am	2010-07-12 17:02:40.000000000 +0200
@@ -83,7 +83,7 @@ expand_content_files=
 # signals and properties.
 # e.g. GTKDOC_CFLAGS=-I$(top_srcdir) -I$(top_builddir) $(GTK_DEBUG_FLAGS)
 # e.g. GTKDOC_LIBS=$(top_builddir)/gtk/$(gtktargetlib)
-GTKDOC_CFLAGS=-I$(NBTK_CFLAGS) -I$(top_srcdir) -I$(top_builddir) -DNBTK_COMPILATION
+GTKDOC_CFLAGS=$(NBTK_CFLAGS) $(GTK_CFLAGS) -I$(top_srcdir) -I$(top_builddir) -DNBTK_COMPILATION
 GTKDOC_LIBS=$(NBTK_LIBS) $(top_builddir)/nbtk/libnbtk-$(NBTK_API_VERSION).la $(top_builddir)/nbtk/libnbtk-gtk-$(NBTK_API_VERSION).la
 
 # This includes the standard gtk-doc make rules, copied by gtkdocize.
diff -up nbtk-1.2.3/docs/reference/libnbtk/nbtk.types nbtk-1.2.3/docs/reference/libnbtk/nbtk
diff -up nbtk-1.2.3/docs/reference/libnbtk/nbtk.types.types nbtk-1.2.3/docs/reference/libnbtk/nbtk.types
--- nbtk-1.2.3/docs/reference/libnbtk/nbtk.types.types	2009-11-27 11:58:44.000000000 +0100
+++ nbtk-1.2.3/docs/reference/libnbtk/nbtk.types	2010-07-12 17:02:40.000000000 +0200
@@ -1,4 +1,6 @@
 #include <nbtk/nbtk.h>
+#include <nbtk/nbtk-subtexture.h>
+#include <nbtk/nbtk-gtk.h>
 
 nbtk_adjustment_get_type
 nbtk_bin_get_type
